### ISSUE TITLE
Don't understand if it works

### ISSUE THREAD

**Issue by eliosh** [LvMalware/cf-proxy#2]
Hi to all.
I start the local proxy with this: 

`bun proxy.js -v -a YOUR-AUTH-TOKEN http test.MYDOMAIN.workers.dev`

Then, in a different terminal, I try: 

`curl -k http://localhost:8080 -H "Authorization: YOUR-AUTH-TOKEN" https://ifconfig.co/'

In the server terminal I can see: 

`[+] Proxying connection to localhost:8080 via test.MYDOMAIN.workers.dev`

but nothing more :-\

The token is the same on the worker deployed in cloudflare

What can I do ?

**Comment by eliosh** [comment]
This is another test via CLI / curl:


`curl 'http://test.MYDOMAIN.workers.dev'   -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7'  -H 'accept-language: it-IT,it;q=0.9,en-US;q=0.8,en;q=0.7'   
 -H 'priority: u=0, i'   
 -H 'sec-ch-ua: "Chromium";v="131", "Not_A Brand";v="24"'   
 -H 'sec-ch-ua-mobile: ?0'   
 -H 'sec-ch-ua-platform: "Linux"'   
 -H 'sec-fetch-dest: document'   
 -H 'sec-fetch-mode: navigate'   
 -H 'sec-fetch-site: none'   
 -H 'sec-fetch-user: ?1'   
 -H 'upgrade-insecure-requests: 1'   
 -H 'Authorization: YOUR-AUTH-TOKEN'
 -H 'Upgrade: websocket'
 -H 'X-Proxy-Target: https://ifconfig.co'
 -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'`

Result is this:
`TypeError: Specified address is missing port.`

I've tried also with ` -H 'X-Proxy-Target: https://ifconfig.co:443'` but it's the same

**Comment by LvMalware** [comment]
It seems that you are just using `curl` the wrong way. Instead of `curl -k http://localhost:8080/ ..`, try `curl -x http://localhost:8080/ ...` ( the right parameter for proxy is `-x` and not `-k`). Also, the auth token doesn't go on Curl, you only pass it when running the proxy.js script with bun like `bun proxy.js -v -a YOUR-AUTH-TOKEN http <yourdomain>.workers.dev`.

Let me know if it works now.

**Comment by LvMalware** [comment]
@eliosh  did it work now?

**Comment by eliosh** [comment]
Hi, thanks for your message.
It works perfectly.

It's normal that it does not work with cloudflare protected website, right?

**Comment by LvMalware** [comment]
@eliosh  Yes. As explained in the readme file, the workers can't connect to addresses that belong to CloudFlare (probably to avoid abuse).

In any case, if your problem is solved, I'm closing this issue.
